#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.3209607305644638 0.5939979201193653 0.7376656968636903 1.264421703452295
  position -1.6398180761368866 -2.799159549891364 3.779739785689448
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Robot {
  translation 0 0 0.025
  children [
    InertialUnit {
      children [
        Solid {
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Box {
                size 0.02 0.02 0.1
              }
            }
          ]
        }
      ]
      name "imu"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.050000000000000024 -0.060000000000000026 0
        rotation 1 -2.994708204376036e-22 0 1.5708
        children [
          DEF wheel Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.666667 0 0
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "wheel4"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.050000000000000024 0.060000000000000026 0
        rotation 1 0 -2.994708204376036e-22 1.5708
        children [
          DEF wheel Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.666667 0 0
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "wheel3"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.050000000000000024 -0.060000000000000026 -7.119062821282043e-19
        rotation 1 1.9626119688198792e-17 3.9252239376397585e-17 1.5707999999999998
        children [
          DEF WHEEL Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.666667 0 0
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "wheel2"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.050000000000000024 0.060000000000000026 0
        rotation 1 2.994708204376036e-22 0 1.5708
        children [
          DEF WHEEL Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.666667 0 0
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    DEF BODY Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 1 0
        }
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
  ]
  name "4_wheel_car"
  boundingObject USE BODY
  physics Physics {
  }
  controller "key_board_controller"
}
